packages/cardano-services/src/Metadata/types.ts

```diff
@@ -1,11 +1,24 @@
 import { Cardano } from '@cardano-sdk/core';
 
-export interface MetadataService {
+export interface DbSyncMetadataService {
   queryTxMetadataByHashes(hashes: Cardano.TransactionId[]): Promise<Map<Cardano.TransactionId, Cardano.TxMetadata>>;
 }
 
+export interface HttpMetadataService {
```

Comment:
I think a better location for new metadata types is src/StakePool

Reply:
Agree, this one is more related to the StakePool context.

> The [interface](https://github.com/input-output-hk/cardano-js-sdk/blob/25bcbef8c529c20211c11589c40bd7f967810bb4/packages/cardano-services/src/Metadata/types.ts#L3) is not coupled to the DB
Yes, so are you saying that having one type

```
interface MetadataService{
   queryTxMetadataByHashes(...)
   getStakePoolExtendedMetadata(...)
}
```
implemented by 2 different services (DbSync and HTTP ) is a preferable way rather than 2 different services, the first located in `/Metadata/` and the second one in `/StakePool/`? And if we have one type then we have to implement both methods by both implementations.

Reply:
> are you saying that having one type:

No, I'm saying to rename the other one back (or to `TxMetadataService`), and the new one to `StakePoolMetadataService` or similar.

> implemented by 2 different services (DbSync and HTTP) is a preferable way rather than having 2 different interfaces

The interface should be source-unaware. The implementation is specific to source, so it can be prefixed with `Http` or `DbSync`.

Reply:
Thx for the clarification, it's addressed with 3c52cbbefd42d95e2a198155d86450ef3ee4fa7b, where https://github.com/input-output-hk/cardano-js-sdk/commit/9e7d7ced0713df8b6d9ab8d06758414e0e5ec403 becomes redundant and will be squashed as well.

Also, I'll classify https://github.com/input-output-hk/cardano-js-sdk/commit/c9295aa8244d5a63c9e3a2d6c632c6b78e4e1ffd as `feat` instead of `refactor` when rebase and squash the commits. @mkazlauskas